### PR TITLE
fix timezone in form and case data pages

### PR DESCRIPTION
**Pull request by dannyroberts**
Before:
screen shot 2015-05-26 at 6 26 34 pm
After:
screen shot 2015-05-26 at 6 27 10 pm

Actual string in XML: `2014-08-06T17:08:03.399-04`. Note, that for pre-migrated domains, i.e. all domains currently, this doesn't actually _use_ the `-04`, it just guesses it from the domain's timezone. That's all about to change :)

~~**Note** This change is unfortunately not strictly an improvement, because a form that was actually submitted with datetimes in UTC will now have those datetimes converted into the domain timezone; and again unfortunately, CloudCare makes all submissions with UTC timestamps. For CloudCare-heavy domains, this will actually make this time incorrect—where it says "EDT" or whatever timezone, it should actually say UTC—but there's no super easy way to detect that.~~

### CODE REVIEW

File: corehq/apps/reports/templatetags/xform_tags.py [hunk 1]
```diff
@@ -15,13 +15,16 @@
 from corehq.apps.reports.formdetails.readable import get_readable_data_for_submission
 from corehq import toggles
 from corehq.util.timezones.conversions import ServerTime
+from corehq.util.timezones.utils import get_timezone_for_user, \
```
Comment (dimagimon) [inline corehq/apps/reports/templatetags/xform_tags.py]: F401 'get_timezone_for_user' imported but unused

File: corehq/apps/reports/templatetags/xform_tags.py [hunk 2]
```diff
@@ -15,13 +15,16 @@
...
+    get_timezone_for_request
+from corehq.util.view_utils import get_request
```
Comment (dimagimon) [inline corehq/apps/reports/templatetags/xform_tags.py]: F401 'get_request' imported but unused

**Comment by dannyroberts** [comment]
Actually hold off on merge—I'm going to do a quick workaround just here that checks if a form is submitted from cloudcare. This is still a problem in other places, and still a problem if you're submitting forms from multiple timezones—but to fix that you'll really have to do the migration.

**Comment by dannyroberts** [comment]
There's no way this failure is related, right?

```
======================================================================
FAIL: testRepeatedFailedLogin (auditcare.tests.auth.authenticationTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "submodules/auditcare-src/auditcare/tests/auth.py", line 115, in testRepeatedFailedLogin
    self.assertEquals(firstlogin_count, next_count)
AssertionError: 1 != 2
----------------------------------------------------------------------
```

File: corehq/util/timezones/utils.py [hunk 3]
```diff
@@ -19,6 +20,19 @@ def reducer(x, y):
...
 
+def get_timezone_for_request(request=None):
+    if request is None:
+        request = get_request()
+
+    user = getattr(request, 'couch_user', None)
+    domain = getattr(request, 'domain', None)
+
+    if user or domain:
+        return get_timezone_for_user(request.couch_user, request.domain)
```
Comment (snopoke) [inline corehq/util/timezones/utils.py]: nitpick but why not just use `user` and `domain` rather than `request.x`

**Comment by snopoke** [comment]
Looks good, still has don't merge label

File: corehq/util/timezones/utils.py (hunk 3 above)
Comment (dannyroberts) [inline corehq/util/timezones/utils.py]: fair enough

**Comment by dannyroberts** [comment]
@kaapstorm buddy

**Pull request merged**